emperor/support_files/js/view-controller.js

```diff
@@ -144,23 +144,39 @@ define([
 
     // there's a few attributes we can only set on "ready" so list them up here
     $(function() {
+      var placeholder = 'Select a ' + scope.title + ' Category';
+
       // setup the slick grid
       scope._buildGrid(options);
 
       scope.refreshMetadata();
 
       // setup chosen
       scope.$select.chosen({width: '100%', search_contains: true,
-                            include_group_label_in_selected: true});
+                            include_group_label_in_selected: true,
+                            placeholder_text_single: placeholder});
 
       // only subclasses will provide this callback
       if (options.categorySelectionCallback !== undefined) {
+
+        // If the number of samples is not too large, fire a callback using the
+        // selector to initialize the data grid
+        if (dm.ids.length < 1000) {
```

Comment:
I think this is an excellent idea but just wondering if we should add a message about it (different behavior) or simply always "start  grayed out", what do you think?